packages/app/app/components/LibraryView/LibraryFolderTree/index.js

```diff
@@ -0,0 +1,166 @@
+      };
+      pathToEntryMap[track.path] = newEntry;
+      folderEntry.children.push(newEntry);
+    }
+
+    const rowEntries = Object.values(pathToEntryMap);
+    const rootEntries = rowEntries.filter(entry => entry.parentId === -1);
+    const idToEntryMap = {};
+    for (const entry of rowEntries) {
+      idToEntryMap[entry.id] = entry;
+    }
+  
+    return {
+      rootEntries,
+      columns: [
+        {
+          title: 'Name',
+          // width: width / 3,
+          width: 0, // required, but flex-grow actually drives the sizing
+          flexGrow: 2,
+          key: 'name',
+          dataKey: 'name'
+        },
+        {
+          title: 'Album',
+          width: 0,
+          flexGrow: 1,
+          key: 'album',
+          dataKey: 'album'
+        },
+        {
+          title: 'Artist',
+          width: 0,
+          flexGrow: 1,
+          key: 'artist',
+          dataKey: 'artist'
+        }
+      ]
+    };
+  }, [localFolders, tracks]);
+};
+
+const LibraryFolderTree = ({
+  tracks,
+  localFolders,
+  expandedFolders
+}) => {
+  const {rootEntries, columns} = useTreeData(tracks, localFolders);
+
+  const tableRef = useRef();
+  const dispatch = useDispatch();
+  const dispatchUpdateExpandedFolders = () => {
+    if (tableRef.current) {
+      dispatch(updateExpandedFolders(tableRef.current.getExpandedRowKeys()));
+    }
+  };
+  useEffect(() => {
+    return dispatchUpdateExpandedFolders;
+  });
+
+  return (
+    <AutoResizer>
+      {({ width, height }) => (
+        <BaseTable
+          ref={tableRef}
+          columns={columns}
+          data={rootEntries}
+          rowHeight={23}
+          expandColumnKey='name'
+          defaultExpandedRowKeys={expandedFolders}
+          onExpandedRowsChange={_.debounce(dispatchUpdateExpandedFolders, 1000)}
+          width={width}
+          height={height}
+          rowRenderer={rowProps => {
+            const {cells, rowData: entry} = rowProps;
+            const key = 'library-track-' + tracks.indexOf(entry.track);
+            if (entry.track) {
+              return (
+                <ContextPopup
+                  key={key}
+                  trigger={<div style={{flex: 1, display: 'flex'}}>{cells}</div>}
+                  thumb={getThumbnail(entry.track)}
+                  title={_.get(entry.track, ['name'])}
+                  artist={_.get(entry.track, ['artist', 'name'])}
+                >
+                  <TrackPopupButtons track={entry.track} withAddToDownloads={false}/>
+                </ContextPopup>
+              );
+            }
+            return (
+              <React.Fragment key={key}>
+                {cells}
+              </React.Fragment>
+            );
+          }}/>
+      )}
+    </AutoResizer>
+  );
+};
+
+LibraryFolderTree.propTypes = {
+  tracks: PropTypes.array,
+  localFolders: PropTypes.arrayOf(PropTypes.string),
+  expandedFolders: PropTypes.arrayOf(PropTypes.string)
+};
+
+export default compose(
```

Comment:
Let's forget about hocs in this component and use `useTranslation` instead, compose is also unneeded.

Reply:
Will do.